blockmanager.go

```diff
@@ -452,9 +447,8 @@ func (b *blockManager) handleTxMsg(tmsg *txMsg) {
 
 	// Process the transaction to include validation, insertion in the
 	// memory pool, orphan handling, etc.
-	allowOrphans := cfg.MaxOrphanTxs > 0
 	acceptedTxs, err := b.txMemPool.ProcessTransaction(tmsg.tx,
-		allowOrphans, true, mempool.Tag(tmsg.peer.ID()))
```

Comment:
I did this for two reasons. 1) This is not the same behavior as Bitcoin Core and 2) it seems odd that we don't send a reject if MaxOrphanTxs is configured to 1 and we already have 1 orphan tx, whereas we do send a reject if MaxOrphanTxs is configured to 0. Essentially, we are saying "this may legitimately be an orphan tx, but we don't want to spend the spend storing it", whereas a reject(duplicate) indicates that we don't think the peer should have broadcast it to us (I think?).